src/array.jl

```diff
@@ -972,3 +972,30 @@ Base.repeat(a::CatArrOrSub{T, N},
 Base.repeat(a::CatArrOrSub{T, N};
             inner = nothing, outer = nothing) where {T, N} =
     CategoricalArray{T, N}(repeat(refs(a), inner=inner, outer=outer), copy(pool(a)))
+
+# JSON3 reading (writing is handled implicitly by definitions
+# in `value.jl`)
+__asmissing__(x::Nothing) = missing
```

Comment:
Doesn't JSON allow representing both `missing` and `nothing`? It's kind of weird that `nothing` would be transformed to `missing` when loading a `CategoricalArray` but not other array types.

Reply:
I don't think so (e.g. [this spec](https://www.json.org/json-en.html)). At least, they are both stored as `null` in JSON3. 

A simple option would be to always treat `null` as `nothing`, and users would be forced to convert `nothing` to `missing` if that's what they want. That sort of makes what is probably the most common use case—reading in a data frame—pretty of annoying.

I can't say I love my solution though, and I do feel iffy about it (hence WIP); If you want to use CategoricalArrays in some other context it might be a surprise that `null` is `missing` just for this one type of array.

@quinnj What do you think? Can you think of a reasonable approach in JSON3 to representing `missing/nothing` differently? 

One option that occurs to me would be to represent small unions using an additional field. So, for example, `a =["x","y","z",missing]` and `b = ["x","y","z",nothing]` would get stored as

```json
{"a": { "data": ["x", "y", "z", ""], "missing": [3]}}
{"b": { "data":["x","y","z",""], "nothing": [3]}}
```

This would then naturally extend to CategoricalArrays.